### DISCUSSION TITLE
Half working Shelly Mini 1 Gen 3

### DISCUSSION THREAD

**Discussion by Xcapee** [RavenSystem/esp-homekit-devices#2293] [MEPLHAA Scripts]
Hello,
I have successfully flashed a Shelly 1 Mini Gen 3, and connected it to HomeKit. When I toggle the light switch, I can see the state change in the Home App perfectly. It's just that the light itself never comes on. I had wired it and successfully tested operation with the original Shelly firmware first, and it was working perfectly. The default MEPLHAA script seems sensible, but I may just be inferring what I want to read, since it's my first time trying to understand the object definitions.

Has anyone else had this happen? Has anyone successfully used the Shelly 1 Mini Gen 3 (not the PM version) with HAA?

Any and all help or pointers much appreciated. Many thanks

**Comment by RavenSystem** [comment]
Please, could you test if built-in LED works? It flashes when you toggle the light switch or turn on/off from Home App.
And check is built/in button works. Pressing it must change state in Home App like your physical switch.
Could you check Shelly temperature from HAA Home Manager (if you have it)?

**Reply by RavenSystem** [reply]
And try this script, please:
```json
{"c":{"io":[[[5,0],2],[[1],6,1],[[10],6,0,1],[[3],10,0,3]],"l":0,"b":[[1,5]]},"a":[{"0":{"r":[[5]]},"1":{"r":[[5,1]]},"b":[[10],[10,0],[1]],"es":[{"t":22,"h":2,"n":5,"g":3,"j":10,"y0":[{"v":75,"r":1,"0":{"m":[[1]]}}]}]}]}
```

**Reply by Xcapee** [reply]
Thanks for this. 

I don't have HAA Home Manager (still trying to work out if this HAA system will work for me - it's my first attempt).

The mini is in my ceiling, so hard to access, and I am just about to leave and won't be home until Thursday this week. 

I did copy/paste the script to the device config page, and pressed "Save" and received an "OK" back. Is it effective immediately? Or do I need to wait for a reboot?

So far there is no change. 

I will get up in the ceiling next weekend and report back then.

**Reply by RavenSystem** [reply]
> Is it effective immediately?

Yes, after few seconds.

I suspect that your Shelly 1 Mini G3 is a new model, and GPIO used by the relay is different. The original Shelly Plus 1 Mini and the Shelly 1 Mini Gen3 use GPIO 7. I changed it to GPIO 5 because other models use it.

You can try different GPIOs:
2, 4, 5, 6, 7, 8, 9, 11, 18, 19, 20 and 21.

Replacing `?` with the GPIO number (there are 3 `?`):
`{"c":{"io":[[[?,0],2],[[1],6,1],[[10],6,0,1],[[3],10,0,3]],"l":0,"b":[[1,5]]},"a":[{"0":{"r":[[?]]},"1":{"r":[[?,1]]},"b":[[10],[10,0],[1]],"es":[{"t":22,"h":2,"n":5,"g":3,"j":10,"y0":[{"v":75,"r":1,"0":{"m":[[1]]}}]}]}]}`

**Reply by Xcapee** [reply]
Yes. That makes perfect sense. I will try when I get back home later in the week.

**Reply by RavenSystem** [reply]
Yesterday, I bought one in Amazon. It arrives today. I hope it is a new model with the unknown GPIO for relay.

**Reply by Xcapee** [reply]
Thank you. That's amazing

**Reply by RavenSystem** [reply]
I have my Shelly device.

Do you remember what firmware version did your Shelly before flashing with HAA?

Did you try to remove power completely from Shelly?

**Reply by RavenSystem** [reply]
Use the original script, because it is right:
`{"c":{"io":[[[7,0],2],[[1],6,1],[[10],6,0,1],[[3],10,0,3]],"l":0,"b":[[1,5]]},"a":[{"0":{"r":[[7]]},"1":{"r":[[7,1]]},"b":[[10],[10,0],[1]],"es":[{"t":22,"h":2,"n":5,"g":3,"j":10,"y0":[{"v":75,"r":1,"0":{"m":[[1]]}}]}]}]}`

And then cut power off of device, and power it again. Device will work perfectly. I tested it with mine, reproducing the issue.

**Reply by RavenSystem** [reply]
Tell me when you try it.

**Reply by Xcapee** [reply]
Hi, I got home tonight from my trip, restored the original script, and power cycled the circuit. I can confirm this worked.

Strange that it needs the power cut once flashed to the new firmware. I wonder why.

Anyway - thanks for your help, support, and passion for this project!

**Reply by RavenSystem** [reply]
Probably, something related to original bootloader and GPIO matrix, because Shelly uses a custom bootloader; and because all its firmware is closed source, then it is impossible to know the specific cause. The only way to be sure that GPIO matrix is reset to factory default is a total power cut.
